infra-sofa-boot-starter/src/main/java/com/alipay/sofa/infra/listener/SofaBootstrapRunListener.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alipay.sofa.infra.listener;
+
+import java.util.*;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.StreamSupport;
+
+import com.alipay.sofa.infra.utils.SOFABootEnvUtils;
+import org.springframework.boot.Banner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.WebApplicationType;
+import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.boot.context.config.ConfigFileApplicationListener;
+import org.springframework.boot.context.event.ApplicationEnvironmentPreparedEvent;
+import org.springframework.context.ApplicationListener;
+import org.springframework.core.Ordered;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.MapPropertySource;
+import org.springframework.core.env.PropertySource;
+import org.springframework.core.env.StandardEnvironment;
+import org.springframework.util.ClassUtils;
+import org.springframework.util.StringUtils;
+
+import com.alipay.sofa.infra.constants.SofaBootInfraConstants;
+
+/**
+ * @author qilong.zql
+ * @since 3.0.0
+ */
+public class SofaBootstrapRunListener implements
+                                     ApplicationListener<ApplicationEnvironmentPreparedEvent>,
+                                     Ordered {
+
+    private final static String  LOGGING_PATH  = "logging.path";
+    private final static String  LOGGING_LEVEL = "logging.level";
+    private static AtomicInteger executed      = new AtomicInteger(0);
+
+    /**
+     * config log settings
+     */
+    private void assemblyLogSetting(ConfigurableEnvironment environment) {
+        if (StringUtils.hasText(environment.getProperty(LOGGING_PATH))) {
+            System.getProperties().setProperty(LOGGING_PATH, environment.getProperty(LOGGING_PATH));
```

Comment:
Should not directly set here consider when LOGGING_PATH is already existed in System Properties.